Realm/RLMSyncSession.h

```diff
@@ -56,6 +105,26 @@ NS_ASSUME_NONNULL_BEGIN
 /// session.
 - (nullable RLMSyncConfiguration *)configuration;
 
-NS_ASSUME_NONNULL_END
+/**
+ Register a progress notification block. Multiple blocks can be registered with the same
+ session at once. Each block will be invoked from the runloop of the thread on which it was
+ registered. The progress notification block will always be called once immediately upon
+ registration in order to provide up-to-date status information.
+
+ The token returned by this method must be retained as long as progress notifications are desired, and the `-stop`
+ method should be called on it when notifications are no longer needed and before the token is destroyed.
+
+ If no token is returned, the notification block will never be called again. There are a number of reasons this might
+ be true. If the session has previously experienced a fatal error it will not accept progress notification blocks.
+ If the block was configured in the `RLMSyncProgressForCurrentlyOutstandingWork` mode but there is no additional
+ progress to report (for example, the number of transferrable bytes and transferred bytes are equal), the block
+ will not be called again.
 
+ @see: `RLMSyncProgressDirection`, `RLMSyncProgress`, `RLMProgressNotificationBlock`, `RLMProgressNotificationToken`
+ */
+- (nullable RLMProgressNotificationToken *)addProgressNotificationBlock:(RLMProgressNotificationBlock)block
```

Comment:
Hmm, I just realized that this API doesn't follow the standard practice of the block being the last parameter to make it easier to use inline blocks. Can we we reorder things so that the block is last?